frontend/app/component/form/polling_station/PollingStationForm.tsx

```diff
@@ -170,3 +239,77 @@ export function PollingStationForm({ electionId, pollingStation, onSaved, onCanc
     </div>
   );
 }
+
+//Generic form validation utils
+
+type ValidationError =
+  | "FORM_VALIDATION_RESULT_REQUIRED"
+  | "FORM_VALIDATION_RESULT_MIN_LENGTH"
+  | "FORM_VALIDATION_RESULT_MAX_LENGTH"
+  | "FORM_VALIDATION_RESULT_INVALID_TYPE"
+  | "FORM_VALIDATION_RESULT_MIN"
+  | "FORM_VALIDATION_RESULT_MAX";
+type FieldRuleBase = {
+  required?: boolean;
+  type: string;
+};
+
+type FieldRuleString = FieldRuleBase & {
+  type: "string";
+  minLength?: number;
+  maxLength?: number;
+};
+
+type FieldRuleNumber = FieldRuleBase & {
+  type: "number";
+  min?: number;
+  max?: number;
+};
+
+type AnyFieldRule = FieldRuleString | FieldRuleNumber;
+type ValidationRules<T> = Record<keyof T, AnyFieldRule>;
+type ValidationResult<T> = Record<keyof T, ValidationError | undefined>;
+
+function validate<T>(rules: ValidationRules<T>, obj: T): [boolean, ValidationResult<T>] {
+  const result: ValidationResult<T> = {} as ValidationResult<T>;
+  Object.entries(rules).forEach(([key, value]) => {
+    const prop = key as keyof T;
+    const inputValue = obj[prop];
+    const rule = value as AnyFieldRule;
```

Comment:
`Object.entries().forEach()` unfortunately does not play very nicely with TypeScript, you could also do
```suggestion
  for (const field in rules) {
    const inputValue = obj[field];
    const rule = rules[field];
```

and then use `continue` instead of `return`

Reply:
this has been refactored